### PR TITLE
test_dots_in_property_names

### PR DESCRIPTION
**Pull request by rohit-rk**

### CODE REVIEW

**Comment by rohit-rk** [comment]
In [variable.to_sql](https://github.com/mozilla/jx-sqlite/blob/master/pyLibrary/queries/expressions.py#L302), acc is acting wierd.
`acc["\\."]["a\\\\.html"]["s"]='"a\\\\.html.$string"'`
is creating `Data({u'.': {u'a.html': {u's': '"a\\\\.html.$string"'}}})`
rather than then `Data({u'.': {u'a\\.html': {u's': '"a\\\\.html.$string"'}}})` .
but when i try this second time, it is works fine.
`Data({u'.': {u'a.html': {u's': '"a\\\\.html.$string"'},
 u'a\\.html': {u's': '"a\\\\.html.$string"'}}})`
Is this expected. `acc = Data()`

**Comment by klahnakoski** [comment]
Looks like you found some very bad logic:

https://github.com/klahnakoski/mo-dots/blob/833b252499db567af308530bdeb3fe28257db4ab/mo_dots/__init__.py#L99

There is also an error in `literal_field()`

https://github.com/klahnakoski/mo-dots/blob/833b252499db567af308530bdeb3fe28257db4ab/mo_dots/__init__.py#L72

it should not be escaping dots that are already escaped.

**Comment by klahnakoski** [comment]
when you say "but when i try this second time, it is works fine." I do not know what you mean by second time.

**Comment by rohit-rk** [comment]
I mean executing it again in debug probe while execution is stopped after this line.

**Comment by klahnakoski** [comment]
add `mo-dots>=1.3.17173` to `requirements.txt` and upgrade mo-dots

**Comment by rohit-rk** [comment]
Done.

**Comment by rohit-rk** [comment]
literal_field is there for escaping dots , but the argument to it is column's proper name, the `\\` there is not to escape dots, so may be we do needed that extra `\\` . I guess it'll be wrong logic then?.

**Comment by klahnakoski** [comment]
You have me confused.  The `\\` being literal depends on the perspective. The `select` clause accepts dot delimited property names: "a\\.html" is referring to the path in `{"a.html":1}`, not the path in `{"a":{"html":1}}`.

**Comment by klahnakoski** [comment]
What is the status of this pull request?  Are you removing the `.replace("\\.", ".")`?

**Comment by rohit-rk** [comment]
Oh yes forgot push it.

**Comment by rohit-rk** [comment]
Literal_field is expecting column names and is escaping dots(as with "a.html"). But in this case here, we have column name (as "a\\.html") . literal_field is taking this as already escaped which is the problem here. May be something like this work:
`acc[literal_field(nested_path)][literal_field(schema.get_column_name(col).replace("\\","\\\\"))][json_type_to_sql_type[col.type]] = quote_column(col.es_column).sql
`

File: jx_sqlite/setop_table.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def place(parent_doc_details):
                                     sql_selects.append(unsorted_sql + " AS " + column_alias)
                                     index_to_column[column_number] = nested_doc_details['index_to_column'][column_number] = ColumnMapping(
                                         push_name=s.name,
-                                        push_column_name=s.name.replace("\\.", "."),
```
Comment (klahnakoski) [inline jx_sqlite/setop_table.py]: This is ok, it is dealing with formatting the output, assuming it is right, given our changes to the tests and to mo-dots

**Comment by klahnakoski** [comment]
rather than patching the problem with `.replace("\\","\\\\")` consider what we are attempting to index.  `acc` is `Data`, which means the property access `[]` will interprets dots as paths. `[literal_field(x)]` is escaping any of those dots so they do not make paths, rather directly access `x`.  Put another way `d[literal_field(x)] == unwrap(d)[x]`.

**Comment by rohit-rk** [comment]
Index needed to be `a\\.html`. but due literal_field 's logic i'm not getting it simply. Is there a way to have an index like this in Data ?

**Pull request merged**